subprojects/core-api/src/main/java/org/gradle/api/artifacts/result/ResolvedComponentResult.java

```diff
@@ -38,23 +38,32 @@
      * <p>The return type is declared as an opaque {@link ComponentIdentifier}, however the identifier may also implement one of the following interfaces:</p>
      *
      * <ul>
-     *     <li>{@link org.gradle.api.artifacts.component.ProjectComponentIdentifier} for those component instances which are produced by the current build.</li>
-     *     <li>{@link org.gradle.api.artifacts.component.ModuleComponentIdentifier} for those component instances which are found in some repository.</li>
+     * <li>{@link org.gradle.api.artifacts.component.ProjectComponentIdentifier} for those component instances which are produced by the current build.</li>
+     * <li>{@link org.gradle.api.artifacts.component.ModuleComponentIdentifier} for those component instances which are found in some repository.</li>
      * </ul>
      *
      * @return the identifier of this component
      */
     ComponentIdentifier getId();
 
+    /**
+     * <p>Returns the identifier of the repository used to source this component.
+     *
+     * @return the identifier of the repository used to source this component , or {@code null} if this component was not resolved from a repository.
+     * @since 4.10
+     */
+    @Nullable
+    String getRepositoryId();
```

Comment:
:o: I'm wondering if this should live on the `ResolvedVariantResult` instead. I think that in the future we would be able to have a component which has variants found in a different repository (our metadata format allows this already). So, this method gives you the repository the _component_ comes from, but the _resolved variant_ can be from another repository. In practice, I think you can keep it here for now, but just be aware that the reality might be more complex in the future.